blog/views.py

```diff
@@ -0,0 +1,58 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.views import generic
+from django.views.generic.edit import FormMixin
+from .forms import CommentaryForm
+
+from blog.models import Post
+
+
+class IndexView(generic.ListView):
+    model = Post
+    posts = Post.objects.all().order_by("-created_time")
+    paginate_by = 5
+    template_name = "blog/index.html"
+    queryset = Post.objects.all().prefetch_related("commentary")
```

Comment:
Defining both 'queryset' and 'model' in a ListView is redundant. If you define 'queryset', the 'model' attribute is not needed. Consider removing one of them.